pygmt/io.py

```diff
@@ -0,0 +1,46 @@
+"""
+PyGMT input/output (I/O) utilities.
+"""
+import xarray as xr
+
+
+def load_dataarray(filename_or_obj, **kwargs):
+    """
+    Open, load into memory, and close a DataArray from a file or file-like
+    object containing a single data variable.
+
+    This is a thin wrapper around :py:meth:`xarray.open_dataarray`. It differs
+    from :py:meth:`xarray.open_dataarray` in that it loads the DataArray into
+    memory, gets GMT specific metadata about the grid via
+    :py:meth:`GMTDataArrayAccessor`, closes the file, and returns the
+    DataArray. In contrast, :py:meth:`xarray.open_dataarray` keeps the file
+    handle open and lazy loads its contents. All parameters are passed directly
+    to :py:meth:`xarray.open_dataarray`. See that documentation for further
+    details.
+
+    Parameters
+    ----------
+    filename_or_obj : str or Path or file-like or DataStore
```

Comment:
I'm not quite sure how to link the `file-like` to https://docs.python.org/3/glossary.html#term-file-like-object. Looking at xarray, they do it at https://github.com/pydata/xarray/blob/f915515d610b4471888fa44dfb00dbae3fd22349/doc/conf.py#L130, so maybe we could add a `napoleon_type_aliases` dictionary to `doc/conf.py`? But we can leave it to a separate PR and just keep it at `file-like` for now.

```suggestion
    filename_or_obj : str or pathlib.Path or file-like or DataStore
```